lib/medusa.ex

```diff
@@ -106,13 +98,12 @@ defmodule Medusa do
         validate_message: &function/3
   """
   def validate_message(functions, %Message{} = message) do
-    global_validator = MedusaConfig.get_message_validator(:medusa_config)
-    functions =
-      cond do
-        is_function(global_validator) -> [global_validator|List.wrap(functions)]
-        true -> List.wrap(functions)
-      end
-    do_validate_message(functions, message)
+    functions = List.wrap(functions)
+    case MedusaConfig.get_message_validator(:medusa_config) do
+      fun when is_function(fun, 1) -> [fun | functions]
+      _ -> functions
```

Comment:
Should we not be more specific on what option causes no function to be added to the validators
i.e.
```
fun when is_function(fun, 1) -> [fun | functions]
nil -> functions
_other -> # some error case